pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -940,6 +953,27 @@ func (r *ReconcileHyperConverged) firstLoopInitialization(request *common.HcoReq
 	r.firstLoop = false
 }
 
+func (r *ReconcileHyperConverged) setOperatorUpgradeableStatus(request *common.HcoRequest) {
+	if hcoutil.GetClusterInfo().IsManagedByOLM() {
+
+		request.Logger.Info("setting the Upgradeable operator condition", requestedStatusKey, !r.upgradeMode)
+
+		msg := hcoutil.UpgradeableAllowMessage
+		status := metav1.ConditionTrue
+		reason := hcoutil.UpgradeableAllowReason
+
+		if r.upgradeMode {
+			msg = hcoutil.UpgradeableUpgradingMessage + r.ownVersion
+			status = metav1.ConditionFalse
+			reason = hcoutil.UpgradeableUpgradingReason
+		}
+
+		if err := r.upgradeableCondition.Set(request.Ctx, status, reason, msg); err != nil {
+			request.Logger.Error(err, "can't set the Upgradeable operator condition", requestedStatusKey, !r.upgradeMode)
```

Comment:
I think we should not simply log this error but return it to the caller and schedule a new reconciliation loop